test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -40,6 +40,31 @@ namespace Npgsql.Tests
 {
     public class ConnectionTests : TestBase
     {
+
+        [OneTimeSetUp]
+        public void CreateTestFile()
+        {
+            // set up pgpass file with connection credentials
+            const string content = "*:*:*:npgsql_tests:npgsql_tests";
+            var pgpassFile = PgPassFile.GetPgPassFilePath();
+            var directory = Path.GetDirectoryName(pgpassFile);
+            if (directory != null)
+            {
+                Directory.CreateDirectory(directory);
+            }
+            File.WriteAllText(pgpassFile, content);
+        }
+
+        [OneTimeTearDown]
+        public void DeleteTestFile()
+        {
+            var pgpassFile = PgPassFile.GetPgPassFilePath();
+            if (File.Exists(pgpassFile))
+            {
+                File.Delete(pgpassFile);
```

Comment:
@rmc00 using the environment variable is a great idea. Restoring the environment variable to whatever it was before the test started (and deleting the mock pgpass file) is a good idea, this way things things are back to exactly what they were. To be extra-careful, please mark these tests with `[Parallelizable(ParallelScope.None)]`, this way NUnit never runs them in parallel (even though parallelization isn't currently on, but it will be at some point).